climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1350 @@
+                lat_lon = firms_cons.iloc[temp][['latitude', 'longitude']].values
+                lat_lon_uni, lat_lon_cpy = np.unique(lat_lon, return_inverse=True, axis=0)
+                cluster_id = DBSCAN(eps=res_data*clus_thres, min_samples=1).\
+                                    fit(lat_lon_uni).labels_
+                cluster_id = cluster_id[lat_lon_cpy]
+                firms_cons.clus_id.values[temp] = cluster_id
+                firms_iter.clus_id.values[firms_iter.event_id == event_id] = \
+                    firms_cons.clus_id.values
+
+        firms.clus_id.values[firms['iter_ev'].values] = firms_iter.clus_id.values
+
+        return firms
+
+    @staticmethod
+    @numba.njit(parallel=True)
+    def _firms_fire(days_thres, fir_cons_id, fir_clus_id, fir_ev_id, fir_iter,
+                    fir_date):
+        """ Creation of event_id for each dataset point.
+        A fire is characterized by a unique combination of 'cons_id' and 'clus_id'.
+
+        Parameters
+        ----------
+            days_thres : int
+                Temporal threshold for clustering
+
+            fir_cons_id : array
+                information on points within the same temporal cluster
+            fir_clus_id : array
+                information on events within the same spatial cluster
+            fir_ev_id : array
+                information of already assigned event IDs
+            fit_iter : array
+                information on itration count
+            fir_date : array
+                date of each data point
+
+        Returns
+        -------
+            firms : pd.DataFrame
+
+        """
+        ev_id = 0
+        for cons_id in np.unique(fir_cons_id):
+            firms_cons = fir_clus_id[fir_cons_id == cons_id]
+            for clus_id in np.unique(firms_cons):
+                fir_ev_id[np.logical_and(fir_cons_id == cons_id, \
+                fir_clus_id == clus_id)] = ev_id
+                ev_id += 1
+
+        for ev_id in np.unique(fir_ev_id):
+            date_ord = np.sort(fir_date[fir_ev_id == ev_id])
+            if (np.diff(date_ord) < days_thres).all():
+                fir_iter[fir_ev_id == ev_id] = False
+            else:
+                fir_iter[fir_ev_id == ev_id] = True
+
+    @staticmethod
+    def _firms_remove_minor_fires(firms, minor_fires_thres):
+        """Remove fires containg fewer FIRMS entries than threshold.
+        A fire is characterized by a unique combination of 'cons_id' and 'clus_id'.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            minor_fires_thres : int
+                threshold of FIRMS data points for an event
+
+        Returns
+        -------
+            firms : pd.DataFrame
+                 FIRMS data
+        """
+        for i in range(np.unique(firms.event_id).size):
+            if (firms.event_id == i).sum() < minor_fires_thres:
+                firms = firms.drop(firms[firms.event_id == i].index)
```

Comment:
I think this half vectorized one-liner might be faster:
see https://stackoverflow.com/questions/48513886/how-can-i-remove-rows-where-frequency-of-the-value-is-less-than-5-python-panda 
```suggestion
       s_event_id = firms['event_id']
        firms[s_event_id.replace(s_event_id.apply(pd.Series.value_counts)).gt(minor_fires_thres).all(1)]
```

Reply:
thanks